docs/source/examples/maxcut-demo.myst

```diff
@@ -0,0 +1,481 @@
+    local_terms = [-0.5 * (np_identity - zz_term) for zz_term in zz_terms]
+    return sum(local_terms)
+    
+```
+
+Since it is diagonal in the computational basis, we can get the ground state energy as follows:
+
+```{code-cell} ipython3
+eig_vals = np.diag(qaoa_hamiltonian(square_graph))
+# Ground state energy
+min(eig_vals)
+```
+
+which, as expected, corresponds to the opposite of the maximum number of cuts of the original graph.
+
+The associated optimal eigenstates are:
+
+```{code-cell} ipython3
+nodes = list({node for edge in square_graph for node in edge})
+num_nodes = len(nodes)
+optimal_bitstrings = [f"{j:b}".zfill(num_nodes) for j, eig_val in enumerate(eig_vals) if eig_val == min(eig_vals)]
+optimal_bitstrings
+```
+
+Which indeed correspond to the 2 optimal bipartitions of the original MaxCut problem:
+
+```{code-cell} ipython3
+[count_cuts_from_string(square_graph, bitstring) for bitstring in optimal_bitstrings]
+```
+
+## Using the Quantum Approximate Optimization Algorithm to find the ground state
+
++++
+
+The size of the previously introduced Hamiltonian is $2^n \times 2^n$. So, when the graph has many nodes,
+finding  the eigenvalues and or the eigenstates as we did above becomes exponentially hard.
+
+**Is there a way of finding the ground state of a Hamiltonian using a near-term quantum computer?**
+
+This is where Quantum Approximate Optimization Algorithm (QAOA)
+[[Farhi, Goldstone, Gutmann, arXiv:1411.4028]](https://arxiv.org/abs/1411.4028) comes into play.
+This method consists of generating a potential solution state through a variational circuit
+$U_{\vec \alpha, \vec \beta}$ which depends on two vectors of $p$ real parameters
+$\vec \alpha = (\alpha_1, .... \alpha_p)$, $\vec \beta= (\beta_1, .... \beta_p)$ and has the following structure:
+ 
+$$|\psi(\vec \alpha, \vec \beta) \rangle = U_{\vec \alpha, \vec \beta}|+\rangle ^{\otimes n} 
+= R_{\alpha_p} W_{\beta_p} \cdots R_{\alpha_1} W_{\beta_1} |+\rangle ^{\otimes n},$$
+
+where $|+> = (|0\rangle + |1\rangle)/\sqrt{2}$ and
+
+$$W_{\beta} = e^{i \beta H }, \quad R_{\alpha} = e^{i \alpha \sum_{j=1}^n X_j },$$
+corresponding to the time evolution generated by system Hamiltonian $H$ and by a mixing Hamiltonian
+$H_{\rm mix}= \sum_j X_j$, respectively.
+
+After applying the circuit to generate $|\psi(\vec \alpha, \vec \beta) \rangle$, one can easily measure
+the energy of the final state
+
+$$E(\vec \alpha, \vec \beta) =  \langle \psi(\vec \alpha, \vec \beta) | H |\psi(\vec \alpha, \vec \beta)\rangle,$$
+
+via simple measurements in the computational basis of the qubits. Repeating this procedures for many
+steps and minimizing the **cost function** $E(\vec \alpha, \vec \beta)$ with respect to the parameters
+$\vec \alpha$ and $\vec \beta$ one can approximate the ground state of $H$.
+
+The maximum number of cuts for the original graph will be given by the opposite of the optimal cost: $- E(\vec \alpha, \vec \beta)^*$.
```

Comment:
```suggestion
The maximum number of cuts for the original graph will be given by the opposite of the optimal cost: $- E(\vec \alpha, \vec \beta)^*$.
```
Is the asterisk here to refer to the optimal $E$? Maybe $- E^*(\vec \alpha, \vec \beta)$ would be more clear.

Reply:
Changed in this [commit]( https://github.com/unitaryfund/mitiq/pull/562/commits/f14806ff05d4b2f76291d347164af95cb2f3bc84).